src/js/base/runtime.js

```diff
@@ -265,7 +264,14 @@ function (Namespace, jsnums, codePoint, seedrandom, util) {
 
     var DefaultReprMethods = {
       "string": String,
-      "number": String,
+      //"number": String,
+      "number": function(val) {
+        if (jsnums.isRoughnum(val)) {
+          return val.toString() + '...';
```

Comment:
Since it's conditionalized on `val` being a roughnum (a properly boxed number and never a primitive double), which is guaranteed to have a `toString()` method (that tacks on the prefix `~`), this is fine.